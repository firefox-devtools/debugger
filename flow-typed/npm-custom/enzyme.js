/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// Using this custom definition of the enzyme module avoids generating some
// flow errors that are unrelated to uses of enzyme itself...
// It would be nice to clean this up.
declare module "enzyme" {
  declare module.exports: any;
}
