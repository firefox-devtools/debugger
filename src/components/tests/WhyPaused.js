import { shallow } from "enzyme";
import renderWhyPaused from "../SecondaryPanes/Frames/WhyPaused.js";

describe("WhyPaused", () => {
  it("should pause reason with message", () => {
    const pause = {
      why: {
        type: "breakpoint",
        message: "bla is hit"
      }
    };
    const component = shallow(renderWhyPaused({ pause }));
    expect(component).toMatchSnapshot();
  });

  it("should show pause reason with exception details", () => {
    const pause = {
      why: {
        type: "exception",
        exception: {
          class: "Error",
          preview: {
            name: "ReferenceError",
            message: "o is not defined"
          }
        }
      }
    };

    const component = shallow(renderWhyPaused({ pause }));
    expect(component).toMatchSnapshot();
  });

  it("should show pause reason with exception string", () => {
    const pause = {
      why: {
        type: "exception",
        exception: "Not Available"
      }
    };

    const component = shallow(renderWhyPaused({ pause }));
    expect(component).toMatchSnapshot();
  });
});
