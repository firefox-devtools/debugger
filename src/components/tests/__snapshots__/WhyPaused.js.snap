// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhyPaused should pause reason with message 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="pane why-paused"
>
    <div>
        Paused on breakpoint
    </div>
    <div
        className="message"
    >
        bla is hit
    </div>
</div>,
  "nodes": Array [
    <div
      className="pane why-paused"
>
      <div>
            Paused on breakpoint
      </div>
      <div
            className="message"
      >
            bla is hit
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <div
    className="pane why-paused"
>
    <div>
        Paused on breakpoint
    </div>
    <div
        className="message"
    >
        bla is hit
    </div>
</div>,
}
`;

exports[`WhyPaused should show pause reason with exception details 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="pane why-paused"
>
    <div>
        Paused on exception
    </div>
    <div
        className="message warning"
    >
        ReferenceError: o is not defined
    </div>
</div>,
  "nodes": Array [
    <div
      className="pane why-paused"
>
      <div>
            Paused on exception
      </div>
      <div
            className="message warning"
      >
            ReferenceError: o is not defined
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <div
    className="pane why-paused"
>
    <div>
        Paused on exception
    </div>
    <div
        className="message warning"
    >
        ReferenceError: o is not defined
    </div>
</div>,
}
`;

exports[`WhyPaused should show pause reason with exception string 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="pane why-paused"
>
    <div>
        Paused on exception
    </div>
    <div
        className="message warning"
    >
        Not Available
    </div>
</div>,
  "nodes": Array [
    <div
      className="pane why-paused"
>
      <div>
            Paused on exception
      </div>
      <div
            className="message warning"
      >
            Not Available
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <div
    className="pane why-paused"
>
    <div>
        Paused on exception
    </div>
    <div
        className="message warning"
    >
        Not Available
    </div>
</div>,
}
`;
